libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/internal/navigation/HttpClient.kt

```diff
@@ -0,0 +1,76 @@
+package com.mapbox.services.android.navigation.v5.internal.navigation
+
+import com.mapbox.navigator.HttpCode
+import com.mapbox.navigator.HttpInterface
+import com.mapbox.navigator.HttpResponse
+import com.mapbox.services.android.navigation.BuildConfig
+import java.io.ByteArrayOutputStream
+import okhttp3.Interceptor
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import okhttp3.logging.HttpLoggingInterceptor
+import okio.Okio
+import timber.log.Timber
+
+internal class HttpClient(
+    private val userAgent: String,
+    private val acceptGzipEncoding: Boolean = false
+) : HttpInterface() {
+
+    private var client: OkHttpClient? = null
```

Comment:
```suggestion
    private val client: OkHttpClient by lazy { clientBuilder.build() }
```

This would allow avoiding the forced cast (`client!!.newCall`) and the nullcheck in `#get`.